plat/renesas/rcar/bl2_plat_setup.c

```diff
@@ -0,0 +1,749 @@
+{
+	u_register_t *boot_kind = (void *) BOOT_KIND_BASE;
+
+	if (is_ddr_backup_mode() == 0)
+		goto cold_boot;
+
+	NOTICE("BL2: skip loading images - resuming from suspend\n");
+	bl31_ep_info->pc = BL31_BASE;
+
+	SET_SECURITY_STATE(bl31_ep_info->h.attr, SECURE);
+	bl31_ep_info->spsr = (uint32_t) SPSR_64(MODE_EL3, MODE_SP_ELX,
+						DISABLE_ALL_EXCEPTIONS);
+	bl2_plat_flush_bl31_params();
+
+	*boot_kind  = RCAR_WARM_BOOT;
+	flush_dcache_range(BOOT_KIND_BASE, sizeof(*boot_kind));
+
+	print_entry_point_info(bl31_ep_info);
+	console_flush();
+
+	bl2_enter_bl31(bl31_ep_info);
+
+cold_boot:
+	*boot_kind  = RCAR_COLD_BOOT;
+	flush_dcache_range(BOOT_KIND_BASE, sizeof(*boot_kind));
+
+	return bl31_ep_info;
+}
+
+meminfo_t *bl2_plat_sec_mem_layout(void)
+{
+	return &bl2_tzram_layout;
+}
+
+bl31_params_t *bl2_plat_get_bl31_params(void)
+{
+	bl2_to_bl31_params_mem_t *params;
+
+	params = (bl2_to_bl31_params_mem_t *) PARAMS_BASE;
+	memset((void *)PARAMS_BASE, 0, sizeof(*params));
+
+	bl2_to_bl31_params = &params->bl31_params;
+	SET_PARAM_HEAD(bl2_to_bl31_params, PARAM_BL31, VERSION_1, 0);
+
+	bl31_ep_info = &params->bl31_ep_info;
+	bl2_to_bl31_params->bl31_image_info = &params->bl31_image_info;
+	SET_PARAM_HEAD(bl2_to_bl31_params->bl31_image_info,
+		       PARAM_IMAGE_BINARY, VERSION_1, 0);
+#ifdef BL32_BASE
+	if (!BL32_BASE)
+		goto done;
+
+	bl2_to_bl31_params->bl32_ep_info = &params->bl32_ep_info;
+	SET_PARAM_HEAD(bl2_to_bl31_params->bl32_ep_info,
+		       PARAM_EP, VERSION_1, 0);
+
+	bl2_to_bl31_params->bl32_image_info = &params->bl32_image_info;
+	SET_PARAM_HEAD(bl2_to_bl31_params->bl32_image_info,
+		       PARAM_IMAGE_BINARY, VERSION_1, 0);
+done:
+#endif
+	bl2_to_bl31_params->bl33_ep_info = &params->bl33_ep_info;
+	SET_PARAM_HEAD(bl2_to_bl31_params->bl33_ep_info,
+		       PARAM_EP, VERSION_1, 0);
+
+	bl2_to_bl31_params->bl33_image_info = &params->bl33_image_info;
+	SET_PARAM_HEAD(bl2_to_bl31_params->bl33_image_info,
+		       PARAM_IMAGE_BINARY, VERSION_1, 0);
+
+	return bl2_to_bl31_params;
+}
+
+void bl2_plat_set_bl31_ep_info(image_info_t * image, entry_point_info_t * ep)
+{
+	SET_SECURITY_STATE(ep->h.attr, SECURE);
+	ep->spsr = SPSR_64(MODE_EL3, MODE_SP_ELX, DISABLE_ALL_EXCEPTIONS);
+}
+
+void bl2_plat_set_bl32_ep_info(image_info_t * image, entry_point_info_t * ep)
+{
+	SET_SECURITY_STATE(ep->h.attr, SECURE);
+	/* The Secure Payload Dispatcher service is responsible for
+	 * setting the SPSR prior to entry into the BL32 image */
+	ep->spsr = 0;
+}
+
+#if (RCAR_BL33_EXECUTION_EL != 0) && (RCAR_BL33_EXECUTION_EL != 1)
+#error
+#endif
+
+#if (RCAR_BL33_EXECUTION_EL == 0)
+#define BL33_MODE MODE_EL1
+#else
+#define BL33_MODE MODE_EL2
+#endif
+
+void bl2_plat_set_bl33_ep_info(image_info_t * image, entry_point_info_t * ep)
+{
+	SET_SECURITY_STATE(ep->h.attr, NON_SECURE);
+	ep->spsr = SPSR_64(BL33_MODE, MODE_SP_ELX, DISABLE_ALL_EXCEPTIONS);
+#ifdef RCAR_BL33_ARG0
+	ep->args.arg0 = RCAR_BL33_ARG0;
+#endif
+}
+
+void bl2_init_generic_timer(void)
```

Comment:
Function bl2_init_generic_timer() used to be called from bl2_reset.S in original Renesas ATF code, see:

func bl2_reset
...
	/* --------------------------------------------
	 * Enable the generic timer via memory mapped
	 * access.
	 * Enable it here quite early in the boot
	 * phase as long as we are still in EL3
	 * --------------------------------------------
	 */
	bl	bl2_init_generic_timer

bl2_reset.S is not present in this implementation and bl2_init_generic_timer() is not being called from anywhere, so this function should be removed from this source file.

Reply:
ack